KNeighbor-yigalron.py

```diff
@@ -0,0 +1,173 @@
+from sklearn.cross_validation import train_test_split
+from sklearn.metrics import roc_auc_score, roc_curve
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.feature_selection import SelectKBest
+
+from sklearn import neighbors
+from sklearn.neighbors import NearestNeighbors
+from statsmodels.robust.scale import mad
+
+
+# In[2]:
+
+n_neighbors = 5
+n_feature_kept = 500
+param_grid = {
+}
+
+
+# In[3]:
+
+get_ipython().magic('matplotlib inline')
+plt.style.use('seaborn-notebook')
+
+
+# In[4]:
+
+# We're going to be building a 'TP53' classifier 
+GENE = 'TP53'
+
+
+# In[ ]:
+
+if not os.path.exists('data'):
+    os.makedirs('data')
+
+
+# In[ ]:
+
+url_to_path = {
+    # X matrix
+    'https://ndownloader.figshare.com/files/5514386':
+        os.path.join('data', 'expression.tsv.bz2'),
+    # Y Matrix
+    'https://ndownloader.figshare.com/files/5514389':
+        os.path.join('data', 'mutation-matrix.tsv.bz2'),
+}
+
+for url, path in url_to_path.items():
+    if not os.path.exists(path):
+        urllib.request.urlretrieve(url, path)
+
+
+# In[5]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('data', 'expression.tsv.bz2')\nX = pd.read_table(path, index_col=0)")
+
+
+# In[6]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('data', 'mutation-matrix.tsv.bz2')\nY = pd.read_table(path, index_col=0)")
+
+
+# In[7]:
+
+y = Y[GENE]
+
+
+# In[ ]:
+
+# The Series now holds TP53 Mutation Status for each Sample
+y.head(6)
+
+
+# In[ ]:
+
+# top samples
+X.head(6)
+
+
+# In[ ]:
+
+# Here are the percentage of tumors with NF1
+y.value_counts(True)
+
+
+# # Set aside 10% of the data for testing
+
+# In[8]:
+
+# Typically, this can only be done where the number of mutations is large enough
+# limit X and y for faster testing
+X_train, X_test, y_train, y_test = train_test_split(X[:3000], y[:3000], test_size=0.1, random_state=0)
+'Size: {:,} features, {:,} training samples, {:,} testing samples'.format(len(X.columns), len(X_train), len(X_test))
+
+
+# ## Create a pipeline to do the prediction
+
+# In[9]:
+
+def fs_mad(x, y):
+    """    
+    Get the median absolute deviation (MAD) for each column of x
+    """
+    scores = mad(x) 
+    return scores, np.array([np.NaN]*len(scores))
+
+# select the top features with the highest MAD
+feature_select = SelectKBest(fs_mad, k=n_feature_kept)
+
+
+# In[14]:
+
+clf = neighbors.KNeighborsClassifier(n_neighbors,'uniform')
+# clf = neighbors.KNeighborsClassifier(n_neighbors,'distance')
+
+# joblib is used to cross-validate in parallel by setting `n_jobs=-1` in GridSearchCV
+# Supress joblib warning. See https://github.com/scikit-learn/scikit-learn/issues/6370
+warnings.filterwarnings('ignore', message='Changing the shape of non-C contiguous array')
+clf_grid = grid_search.GridSearchCV(estimator=clf, param_grid=param_grid, n_jobs=-1, scoring='roc_auc')
+pipeline = make_pipeline(
+    feature_select,  # Feature selection
```

Comment:
I think we should remove the MAD feature selection. We don't really need to save time, since you're code is running quickly. I think it could help performance to have more features to determine "nearness".